src/main/java/org/zalando/nakadi/controller/SchemaController.java

```diff
@@ -0,0 +1,37 @@
+package org.zalando.nakadi.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.zalando.nakadi.service.Result;
+import org.zalando.nakadi.service.SchemaService;
+import org.zalando.problem.spring.web.advice.Responses;
+
+@RestController
+public class SchemaController {
```

Comment:
My vision of tests in Nakadi is different.
I think you should have only one very straightforward acceptance test for this endpoint and you should have a lot of unit tests for controller to test all corner cases.
I think this is the approach we agreed on some time ago. To test everything that is possible with a unit tests and to have very few acceptance tests. To follow the approach of test pyramid.

(The only exception I see are the cases when testing with unit test will lead to mock-nightmare and testing with AT is very simple - then AT is preferable)